PrestaShop/ps_googleanalytics
# Add tracking of cancelled orders

This PR enables tracking of cancelled orders, which could greatly affect the statistics. The refund functionality is already provided by the enhanced ecommerce library, but never used.

I added a configuration field to module settings, where the user can select in which order states, he wants the order to be refunded (cancelled) in Google Analytics.

When the order status changes to one of these, it sends the information to google analytics.

**Technical changes**
- The functionality hooks to actionOrderStatusPostUpdate.
- Adds a configuration field and an entry in database with encoded order states.
- Adds 1 column to ps_ganalytics to prevent multiple refunds for one order. 
- Bumps module version to 4.1.0 to run the upgrade script needed - to upgrade the database and insert a configuration field.
- Fixes forgotten echo and cookie->write in hookDisplayAdminOrder.

**How it works**
- After you change the order status, the module checks, if the new status is one of the states the user wants the order to be cancelled.
- If so, it checks the database, if this order was not cancelled before - to prevent multiple refunds, which Google Analytics unfortunately allows.
- If no refund information found in database, adds the refund action by MBG.refundByOrderId.

**How to test**
1. Install Google Tag Assistant (in Chrome).
2. Install ps_googleanalytics module, set it up. Add some states to the configuration field.
3. Make an order, go to it's administration.
4. Enable recording in Google Tag Assistant and refresh page.
5. Change order status to cancelled.
6. After page refresh, check Google Tag Assistant and see the Refund event being sent.
7. Change order status to "Processing", then, change it back to "Cancelled".
8. No refund event should be sent.

The refund will appear after minutes/hours in "Refund" column in Google Analytics.
